### PR TITLE
Shortening of long lists with a x remaining message and popup#90

### PR DESCRIPTION
**Pull request by theisuru**
ebi-ait/data-catalogue#90

Whats in this pull request
- React class component to render more list values in a cell
- Render all array type values in the gird using this component

Points to discuss
- Naming is not very good any suggested name for this component?
- Currently this will render any array value in the schema, becaues of the way ag grid is defined from `schema`. This needs to be discussed. 
- UI and design

### CODE REVIEW

**Comment by sandsebi** [comment]
Looks good to me. Maybe we should set up [Github codespaces](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces#monthly-included-storage-and-core-hours-for-personal-accounts) so we can see an actual live version of each PR, it works similar to CodeSandBox. But seems to be free for 120 hours per month.

File: src/theme/theme.css [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
     --color-main: #15a2e7;
     --color-secondary: #deafe5;
+    --color-background: #234345;
+
+    --color-font-primary: #000000;
+    --color-font-secondary: #444444;
+    --color-font-link: #6a72cb;
 
     --font-main: "DejaVu Sans", serif;
 }
```
Comment (amnonkhen) [inline src/theme/theme.css:11]: 👍

File: src/MoreCellRenderer/MoreCellRenderer.module.css [hunk 2]
```diff
@@ -0,0 +1,41 @@
+@import '../theme/theme.css';
+
+.MoreCellWrapper {
```
Comment (amnonkhen) [inline src/MoreCellRenderer/MoreCellRenderer.module.css:3]: class names usually start with a lowercase letter

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/MoreCellRenderer/MoreCellRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                            vertical: 'bottom',
+                            horizontal: 'left',
+                        }}
+                        className={style.MorePopper}
+                    >
+                        <div className={style.MorePopoverTitle}>{value.length} values</div>
+                        <div className={style.MorePopoverValues}>{value.map((val => <div
+                            className={style.MorePopoverValue}
+                            key={val}>{val}</div>))}</div>
+                    </Popover></div>}
```
Comment (amnonkhen) [inline src/MoreCellRenderer/MoreCellRenderer.tsx:59]:
```suggestion
                    </Popover>
                </div>}
```

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: src/Catalogue.tsx [hunk 4]
```diff
@@ -30,6 +31,7 @@ const Catalogue: React.FC<CatalogueProps> = ({schema}) => {
                         console.log(params.newValue)
                         return params.newValue.split(',');
```
[the remark this answers is not in the document]
Comment (theisuru) [reply src/Catalogue.tsx:32]: actually we don't need this code segment. I will remove it.

File: src/MoreCellRenderer/MoreCellRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import style from "./MoreCellRenderer.module.css";
+import Popover from '@mui/material/Popover';
+
+interface MoreCellRendererState {
+    anchorEl: HTMLDivElement | null;
+}
+
+export default class MoreCellRenderer extends React.Component<{ value: string[] }, MoreCellRendererState> {
```
[the remark this answers is not in the document]
Comment (theisuru) [reply src/MoreCellRenderer/MoreCellRenderer.tsx:9]: This is not a plural value. It is the div element that is holding the popover

**Comment by theisuru** [comment]
Renamed to `ListCellRenderer". I think it would make sense to have all `cell renderer` with a same naming convention and inside a one directory, later when we have more cell renderer types.

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**